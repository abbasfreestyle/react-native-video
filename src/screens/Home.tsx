import React, { FC } from 'react';
import { Button } from 'react-native';

import { useNavigation } from '@react-navigation/native';

import { NavigationProps } from '../navigation/types';

export const Home: FC = () => {
  const navigation = useNavigation<NavigationProps<'Home'>>();

  return (
    <>
      <Button
        title="Scrolled View"
        onPress={() => navigation.navigate('Scrolled View')}
      />
      <Button
        title="Tabbed View"
        onPress={() => navigation.navigate('Tabbed Views')}
      />
    </>
  );
};
