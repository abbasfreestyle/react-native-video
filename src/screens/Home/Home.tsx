import React from 'react';
import { Button } from 'react-native';

import { useNavigation } from '@react-navigation/native';

import { NavigationProps } from '../../navigation/types';

export const Home = () => {
  const navigation = useNavigation<NavigationProps<'Home'>>();

  return (
    <>
      <Button
        title="Scrolled View"
        onPress={() => navigation.navigate('Scrolled View')}
      />
      <Button
        title="Tabbed Views"
        onPress={() => navigation.navigate('Tabbed Views')}
      />
      <Button
        title="Storybook"
        onPress={() => navigation.navigate('Storybook')}
      />
    </>
  );
};
