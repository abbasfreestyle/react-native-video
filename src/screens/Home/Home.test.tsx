import React from 'react';

import { NavigationContainer, useNavigation } from '@react-navigation/native';
import { act, fireEvent, render } from '@testing-library/react-native';

import { Home } from './Home';

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: jest.fn(),
}));

describe('Home Screen', () => {
  const cases = [['Scrolled View'], ['Tabbed Views'], ['Storybook']];
  test.each(cases)('%s can navigate to the expected screen', (button) => {
    (useNavigation as jest.Mock).mockReturnValue({
      navigate: jest.fn(),
    });
    const { navigate } = useNavigation();

    const { getByText } = render(<Home />, { wrapper: NavigationContainer });

    const btn = getByText(button);
    act(() => fireEvent.press(btn));

    expect(navigate).toHaveBeenCalledWith(button);
  });

  test('should match the snapshot', () => {
    const { toJSON } = render(<Home />, { wrapper: NavigationContainer });

    expect(toJSON()).toMatchSnapshot();
  });
});
