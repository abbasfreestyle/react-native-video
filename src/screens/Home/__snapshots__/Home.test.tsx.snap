// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen should match the snapshot 1`] = `
Array [
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Scrolled View
      </Text>
    </View>
  </View>,
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Tabbed Views
      </Text>
    </View>
  </View>,
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Storybook
      </Text>
    </View>
  </View>,
]
`;
